bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java

```diff
@@ -992,6 +1050,105 @@ private void handleToneCmd(int current, String channel, Command command, int nbS
         }
     }
 
+    /**
+     * Handle a tcbypass command
+     *
+     * @param onlyToggle true if only the toggle command must be used
+     * @param channel the channel
+     * @param command the received channel command (OnOffType)
+     * @param onCmd the command to be sent to the device to bypass_on
+     * @param offCmd the command to be sent to the device to bypass_off
+     * @param toggleCmd the command to be sent to the device to toggle the bypass state
+     *
+     * @throws RotelException in case of communication error with the device
+     */
+    private void handleTcbypassCmd(boolean onlyToggle, String channel, Command command, RotelCommand onCmd,
+            RotelCommand offCmd, RotelCommand toggleCmd) throws RotelException, InterruptedException {
+        if (command instanceof OnOffType) {
+            if (onlyToggle) {
+                connector.sendCommand(toggleCmd);
+            } else if (command == OnOffType.ON) {
+                connector.sendCommand(onCmd);
+            } else if (command == OnOffType.OFF) {
+                connector.sendCommand(offCmd);
+                Thread.sleep(100);
+                connector.sendCommand(RotelCommand.BASS);
+                Thread.sleep(100);
+                connector.sendCommand(RotelCommand.TREBLE);
```

Comment:
You request bass/trebble after enabling tone controls.
I imagined that the reason is that bass/trebble are set to 0 when bypass is ON ?
Why do you not do the same thing when bypass is set to ON ?

Reply:
Last opened comment

Reply:
as you suggests:
in case TCBYPASS is ON:
both TREBLE and BASS are 0 (rather undefined, because "bypassed"; but the device's answer is 0)
so you are right: we can ask ("treble?"/"bass?"); we will get answer="0"
in case TCBYPASS is OFF
both TREBLE and BASS have their former values (set before TCBYPASS was going to ON)
when we ask ("treble?"/"bass?") we will get answer="former value"

you are right: it is correct to make the same in both cases

Reply:
another comment:
In the case of TCBYPASS=ON, the possibility of setting BASS/TRBLE is suppressed by the unit; related activities on the remote control are ignored. In this case, it is pointless to offer BASS/TREBLE as an item on a user interface. In my simple user interface, I have added the following option to the BASS/TREBLE item as a dependency on item TCBYPASS:

component: oh-slider-card
config:
  item: A14_Bass
  ....
  title: Bass
  visible: =items.A14_TCBypass.state=="OFF"
  
But this is not part of the handler code

Reply:
Yes, you should ignore commands for bass and trebble channels in handleCommand when variable tcbypass is true. More precisely, please run the command to get the current value to restore the right value when tcbypass is on.

Reply:
For my own understanding:

In state TCBYPASS OFF
   the user is enabled to adjust BASS/TREBLE ("bass_+05!")
   the device modifies appropriate settings in their memory (set to +5)
   the device responses newly adjusted values automatically ("bass_+05$")
   the user hears/sees (on his GUI) what he gets from the device

In state TCBYPASS ON
   the device ignores adjustments of BASS/TREBLE  ("bass_-08!")
   the device doesn't automatically response any adjustment activity ("silence" is returned)
   the device does bypass user's BASS/TREBLE adjustments
   and uses somehow calculated BASS/TREBLE adjustments instead
   
   BUT
     retrieving programmatically values for BASS/TREBLE the device returns the memory-values that are currently not used at all  ("bass_+05$")
   
   that means in state TCBYPASS ON the user doesn't get/hear what it sees on his GUI;
   the device outputs BASS/TREBLE values for the OFF state while using some optimized values for the ON state

Reply:
my suggestion (see last commit):

1) switching TCPYPASS into ON state in function handleTcbypassCmd():
   set programmatically class variables [bass]/[treble] = 0
   updateChannelState(BASS/TREBLE)
   so in the case of doing this switch the user observes in GUI a change on BASS/TREBLE (from +5 to 0)

2) switching TCPYPASS into OFF state in function handleTcbypassCmd():
   set programmatically class variables [bass]/[treble] to device's memory values
   updateChannelState(BASS/TREBLE)
   so in the case of doing this switch the user observes in GUI a change on BASS/TREBLE (from 0 to +5)

3) in function handleCommand():
   user's adjustments ("bass_-08!") on BASS/TREBLE are ignored in TCPYPASS ON state
   class variables [bass]/[treble] have been set programmatically to 0 (see above in handleTcbypassCmd())
   updateChannelState(BASS/TREBLE) resets BASS/TREBLE (from -8 to 0)
   so user's effort to adjust BASS/TREBLE will be "ignored"
   
   that means in state TCBYPASS ON the user now sees always BASS/TREBLE is 0 (UNDEF would be better)
   whereas in TCBYPASS OFF he sees (and hears) values adjusted by himself.

Reply:
is it possible to disable/enable programmatically a channel or item???
info regarding the work schedule: i will not be able to work for the next 9 days

Reply:
> is it possible to disable/enable programmatically a channel or item???

No

> info regarding the work schedule: i will not be able to work for the next 9 days

Is it possible to finish today, or is it already too late for you ?